tests/Unit/Helpers/Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/TestHelpers.hpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Framework/TestingFramework.hpp"
+
+#include <memory>
+#include <pup.h>
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/DampingFunction.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Overloader.hpp"
+
+namespace TestHelpers::GeneralizedHarmonic::ConstraintDamping {
+namespace detail {
+template <size_t VolumeDim, typename Fr, class... MemberArgs, class T>
+void check_impl(
+    const std::unique_ptr<
+        ::GeneralizedHarmonic::ConstraintDamping::DampingFunction<
+            VolumeDim, Fr>>& in_gh_damping_function,
+    const std::string& python_function_prefix, const T& used_for_size,
+    const std::array<std::pair<double, double>, 1> random_value_bounds,
+    const MemberArgs&... member_args) noexcept {
+  using GhDampingFunc =
+      ::GeneralizedHarmonic::ConstraintDamping::DampingFunction<VolumeDim,
+                                                                  Fr>;
+  using CallOperatorFunction =
+      Scalar<T> (GhDampingFunc::*)(const tnsr::I<T, VolumeDim, Fr>&)
+          const noexcept;
+
+  const auto member_args_tuple = std::make_tuple(member_args...);
+  const auto helper =
+      [&python_function_prefix, &random_value_bounds, &member_args_tuple,
+       &used_for_size](
+          const std::unique_ptr<GhDampingFunc>& gh_damping_function) noexcept {
+        // need func variable to work around GCC bug
```

Comment:
Is there documentation of what this gcc bug is and why one gets around it in this way? Does it work to say `auto func = &GhDampingFunc::operator();` (so you wouldn't need to explicitly declare CallOperatorFunction)?

Reply:
There is no documentation. This comment and the explicit CallOperatorFunction declaration originally appeared in #625 , by @nilsdeppe , three years ago, in ` tests/Unit/PointwiseFunctions/EquationsOfState/TestHelpers.hpp:64` in April 2018. I kept the comment when adding an analogous `tests/Unit/Helpers/PointwiseFunctions/MathFunctions/TestHelpers.hpp`, and again when adding this `TestHelpers.hpp` for constraint damping.

@nilsdeppe do you recall what the bug was? If so, maybe document in a comment on this PR?

If you both insist, I'm happing to change this to Mark's suggestion or to whatever else you prefer, but unless there's a good reason to change it, why not leave it as is for expediency's sake?

Reply:
I have no idea. I would try removing it and seeing if it passes the builds. Given how old the original commit is, it could very well be for a version of GCC we don't use anymore.